src/aggregator/aggregation/resend/resend_buffer.go

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package resend
+
+import (
+	"math"
+	"time"
+
+	"github.com/uber-go/tally"
+)
+
+const bufferLimit = time.Minute
+
+// Buffer is a fixed-size buffer for servicing resends of min and max
+// aggregations.
+type Buffer interface {
```

Comment:
is there a reason for the interface vs just a plain, public struct? since it has 0 dependencies/unlikely to have alternative interfaces, it would result in less indirection to pass *Buffer around, enable inlining of Buffer calls, and not break any clients if a new method is added.

Reply:
s/alternative interfaces/alternative implementations/